subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -988,16 +969,10 @@ TestFramework useTestFramework(TestFramework testFramework) {
     }
 
     private <T extends TestFrameworkOptions> TestFramework useTestFramework(TestFramework testFramework, @Nullable Action<? super T> testFrameworkConfigure) {
+        final TestFramework oldFramework = Test.this.testFramework.get();
+
         if (optionsAccessed) {
-            DeprecationLogger.deprecateAction("Accessing test options prior to setting test framework")
-                .withContext("\nTest options have already been accessed for task: '" + getProject().getName() + ":" + getName() + "' prior to setting the test framework to: '" + testFramework.getClass().getSimpleName() + "'. Any previous configuration will be discarded.\n")
-                .willBeRemovedInGradle8()
-                .withDslReference(Test.class, "options")
-                .nagUser();
-
-            if (!this.testFramework.get().getClass().equals(testFramework.getClass())) {
-                getLogger().warn("Test framework is changing from '{}', previous option configuration would not be applicable.", this.testFramework.get().getClass().getSimpleName());
-            }
+            throw new GradleException("Cannot set test framework after accessing test options.  Framework was previously: " + oldFramework.getClass().getSimpleName() + ", attempting to set: " + testFramework.getClass().getSimpleName() + ".");
```

Comment:
I think we should use a prettier name here than the test framework class names.  I don't think these are that visible to a user.

Maybe something like
> You cannot change the test framework to TestNG after accessing test options.  The current test framework is JUnit4.